app/main.py

```diff
@@ -1,2 +1,54 @@
+from copy import copy
+from typing import Hashable, Any
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.capacity = 8
+        self.threshold = int(self.capacity * 2 / 3)
+        self.hash_table = [[]] * self.capacity
+        self.size = 0
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        if self.size == self.threshold:
+            self.resize()
+
+        hash_key = hash(key)
+        index = hash_key % self.capacity
+
+        while True:
+            if not self.hash_table[index]:
+                self.hash_table[index] = [key, value, hash_key]
+                self.size += 1
+                break
+
+            if self.hash_table[index][0] == key:
+                self.hash_table[index][1] = value
+                break
+
+            index = (index + 1) % self.capacity
+
+    def __getitem__(self, item: Hashable) -> Any:
+        index = hash(item) % self.capacity
+
+        for _ in range(len(self.hash_table)):
+            if self.hash_table[index] and self.hash_table[index][0] == item:
```

Comment:
You should immediately raise `KeyError` when `self.hash_table[index]` is `[]`, otherwise it's not a constant time operation while it should be.

Reply:
You're right